clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3227,6 +3256,46 @@ private void deserializeValue(
     }
   }
 
+  private GenericRecord deserializeKeyAndReturn(
```

Comment:
I don't think we need two separate functions for key and value.  I think we can just add an argument for SchemaEntry and just invoke it with:

deserializeAvroObjectAndReturn(schemaId, object, record, schemaRepository.getValueSchema(storeName, schemaId)) (and change it to getKeySchema for the other call).

Reply:
One other thing, let's a TODO here.  Something I 'think' is the case, this code will not work if the data which is ingested is compressed or chunked.

Now, compression or chunking isn't ALWAYS enabled for a store.  In fact, it's probably in the minority.  But we'll need to do some work here I think to make this work with those features.